File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.swift
import Foundation

func solution(_ X:String, _ Y:String) -> String {
    var countX: [Int] = Array(repeating: 0, count: 10) // X에서 숫자 빈도
    var countY: [Int] = Array(repeating: 0, count: 10) // Y에서 숫자 빈도
    var result = ""
    X.forEach { countX[Int(String($0))!] += 1 }
    Y.forEach { countY[Int(String($0))!] += 1 }
    for i in stride(from: 9, through: 0, by: -1) {
        let commonCount = min(countX[i], countY[i])
        result += String(repeating: "\(i)", count: commonCount)
    }
    
    if result.isEmpty { 
        return "-1" 
    } else {
        return result.first! == "0" ? "0" : result
    }
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int arrX[10];
int arrY[10];

string solution(string X, string Y) {
    string answer = "";

    for (char ch : X) {
        arrX[ch - '0'] += 1;
    }

    for (char ch : Y) {
        arrY[ch - '0'] += 1;
    }

    for (int i = 9; 0 <= i; i--) {
        int count = min(arrX[i], arrY[i]);

        for (int j = 0; j < count; j++) {
            answer += to_string(i);
        }
    }

    return (answer.empty()) ? "-1" : (answer[0] == '0') ? "0" : answer;
}